doc/pages/architecture/080-mqtt-v1-protocol.md

```diff
@@ -0,0 +1,278 @@
+0000040 00 00
+```
+
+```
+0000000 0c 00 00 00 10 76 00 0a 00 00 00 00
+```
+
+```
+0000000 09 00 00 00 08 76 00 01 00
+```
+
+Property messages order must be preserved and they must be consumed in order. The same property with the same value can be sent several times, this behavior is allowed but discouraged: it's up to the device to avoid useless messages.
+A device must also make sure to publish all the properties that have been changed while the device was offline.
+
+#### Unset Property Message
+
+Properties can be unset with an unset message. An unset message is just an empty 0 bytes payload.
+
+#### Datastream Message (individual aggregation)
+
+Datastream messages for interfaces with individual aggregation have a "v" key and an optional "t" key (which means timestamp) and they look like the followings:
+
+* `{"v": false}`
+
+* `{"v": 16.73}`
+
+* `{"v": 16.73, "t": 1537449422890}`
+
+Timestamps are UTC timestamps (BSON 0x09 type), when not provided reception timestamp is used.
+
+Previous payloads are in order BSON encoded as following hex dumps:
+
+```
+0000000 09 00 00 00 08 76 00 00 00
+```
+
+```
+0000000 10 00 00 00 01 76 00 7b 14 ae 47 e1 ba 30 40 00
+```
+
+```
+0000000 1b 00 00 00 09 74 00 2a 70 20 f7 65 01 00 00 01
+0000020 76 00 7b 14 ae 47 e1 ba 30 40 00
+```
+
+#### Datastream Message (object aggregation)
+
+Datastream messages (for interfaces with object aggregation) have a "v" key, that is a BSON subdocument, and an optional "t" key (which means timestamp) and they look like the following:
+
+* `{"v": {"temp": 25.3123, "hum": 67.112}}`
+
+* `{"v": {"temp": 25.3123, "hum": 67.112}, "t": 1537452514811}`
+
+Timestamps are UTC timestamps (BSON 0x09 type), when not provided reception timestamp is used.
+
+Previous payloads are in order BSON encoded as following hex dumps:
+
+```
+0000000 28 00 00 00 03 76 00 20 00 00 00 01 68 75 6d 00
+0000020 ba 49 0c 02 2b c7 50 40 01 74 65 6d 70 00 72 8a
+0000040 8e e4 f2 4f 39 40 00 00
+```
+
+```
+0000000 33 00 00 00 09 74 00 fb 9d 4f f7 65 01 00 00 03
+0000020 76 00 20 00 00 00 01 68 75 6d 00 ba 49 0c 02 2b
+0000040 c7 50 40 01 74 65 6d 70 00 72 8a 8e e4 f2 4f 39
+0000060 40 00 00
+```
+
+## Minimal Protocol
+
+A device might implement a subset of this protocol if needed. `/control/consumer/properties`, `/control/producer/properties` and `/emptyCache` might be ignored or not implemented if a device has no property interfaces.
+A further simplification might remove any requirement for any introspection message when previously provisioned, but this feature is not supported out of the box.
+
+## Error Handling
+
+A device might be forcefully disconnected due to any kind of error. Devices should wait a random ammount of time before trying to connect again to the broker.
+_session present_ might be also set to false to ensure a clean and consistent state (in that case messages such as _introspection_ and _empty cache_ should published as previously described).
```

Comment:
asterisks

Reply:
_foo_ and *foo* are the same, maybe _ is more common for italic.